test/fixture/infinity-stack-trace.js

```diff
@@ -0,0 +1,14 @@
+import test from '../../';
+import {AssertionError} from '../../lib/assert';
+
+test(t => {
+	const limitBefore = Error.stackTraceLimit;
+	Error.stackTraceLimit = 1;
+	try {
+		throw new AssertionError({});
+	} catch (err) {
+		console.error(err.stack);
```

Comment:
Rather than instantiating the error, call say `t.fail()` from inside some nested functions:

```js
const a = () => b();
const b = () => c();
const c = () => t.fail();
```

AVA should print the full stack trace which you can then assert against in `test/cli.js`.

Reply:
updated